django_celery_beat/models.py

```diff
@@ -314,21 +314,26 @@ class Meta:
 
     def validate_unique(self, *args, **kwargs):
         super(PeriodicTask, self).validate_unique(*args, **kwargs)
-        if not self.interval and not self.crontab and not self.solar:
+        schedule_info = {'interval': self.interval,
+                         'crontab': self.crontab,
+                         'solar': self.solar}
+        selected_schedule_types = [schedule_type
+                                   for schedule_type in schedule_info
+                                   if schedule_info[schedule_type]]
```

Comment:
maybe something more obvious like `selected_schedule_types = [s for s in ['interval', 'crontab', 'solar'] if getattr(self, s)]`?

Reply:
> maybe something more obvious like `selected_schedule_types = [s for s in ['interval', 'crontab', 'solar'] if getattr(self, s)]`?

yes, more clear and beautiful. update it !